app/src/main/java/org/covidwatch/android/data/signedreport/SignedReport.kt

```diff
@@ -0,0 +1,117 @@
+package org.covidwatch.android.data.signedreport
+
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import androidx.room.TypeConverters
+import org.covidwatch.android.data.DateConverter
+import org.covidwatch.android.data.SignedReportUploadStateConverter
+
+@Entity(tableName = "signed_reports")
+@TypeConverters(DateConverter::class, SignedReportUploadStateConverter::class)
+data class SignedReport(
+    @ColumnInfo(name = "end_index")
+    var endIndex: Int = 0,
+
+    @ColumnInfo(name = "is_processed")
+    var isProcessed: Boolean = false,
+
+    @ColumnInfo(name = "memo_data", typeAffinity = ColumnInfo.BLOB)
+    var memoData: ByteArray = ByteArray(0),
+
+    @ColumnInfo(name = "memo_type")
+    var memoType: Int = 0,
+
+    @ColumnInfo(name = "report_verification_public_key_bytes", typeAffinity = ColumnInfo.BLOB)
+    var reportVerificationPublicKeyBytes: ByteArray = ByteArray(0),
+
+    @PrimaryKey
+    @ColumnInfo(name = "signature_bytes", typeAffinity = ColumnInfo.BLOB)
+    var signatureBytes: ByteArray = ByteArray(0),
+
+    @ColumnInfo(name = "start_index")
+    var startIndex: Int = 0,
+
+    @ColumnInfo(name = "temporary_contact_key_bytes", typeAffinity = ColumnInfo.BLOB)
+    var temporaryContactKeyBytes: ByteArray = ByteArray(0),
+
+    @ColumnInfo(name = "upload_state")
+    var uploadState: UploadState = UploadState.NOTUPLOADED
+
+) : Parcelable {
```

Comment:
how about `@Parcelize`?

Reply:
Maybe :)
I don't know, in the first place, if we need this class to be serializable. I'll check it out, thanks :)